tests/resolve_method.rs

```diff
@@ -0,0 +1,148 @@
+use jst::{convert_avro, convert_bigquery};
+use jst::{Context, ResolveMethod};
+use serde_json::Value;
+
+#[test]
+fn test_bigquery_object_error_resolution() {
+    let mut expected: Value;
+    let mut context: Context;
+
+    let input: Value = serde_json::from_str(
+        r#"
+        {
+            "type": "object",
+            "properties": {
+                "empty": {},
+                "int": {"type": "integer"}
+            }
+        }
+        "#,
+    )
+    .unwrap();
+
+    expected = serde_json::from_str(
+        r#"
+        [
+            {
+                "mode": "NULLABLE",
+                "name": "empty",
+                "type": "STRING"
+            },
+            {
+                "mode": "NULLABLE",
+                "name": "int",
+                "type": "INT64"
+            }
+        ]
+        "#,
+    )
+    .unwrap();
+
+    context = Context {
+        resolve_method: ResolveMethod::Cast,
+    };
+    assert_eq!(expected, convert_bigquery(&input, context));
+
+    expected = serde_json::from_str(
+        r#"
+        [
+            {
+                "mode": "NULLABLE",
+                "name": "int",
+                "type": "INT64"
+            }
+        ]
+        "#,
+    )
+    .unwrap();
+    context = Context {
+        resolve_method: ResolveMethod::Drop,
+    };
+    assert_eq!(expected, convert_bigquery(&input, context));
+
+    context = Context {
+        resolve_method: ResolveMethod::Panic,
+    };
+    assert!(std::panic::catch_unwind(|| convert_bigquery(&input, context)).is_err());
```

Comment:
Why not split these multiple tests that are currently in one function into individual tests functions? You could then use `#[should_panic]` on the functions.

I know this means duplicating the `input` (or making that a constant), but IMO that's cleaner and it's easier to debug a single failing test function.

Reply:
Refactored. Debugging actually wasn't too bad in my workflow, since the tests naturally progressed in order of casting, dropping, and panicking. The test fixture had to be a function since "calls in constants are limited to constant functions, tuple structs and tuple variants".